code/object/objectdock.cpp

```diff
@@ -733,6 +831,38 @@ void object_remove_arriving_stage2_ndl_flag_helper(object *objp, dock_function_i
 		Ships[objp->instance].flags.remove(Ship::Ship_Flags::Arriving_stage_2_dock_follower);
 }
 
+void dock_calc_total_moi_helper(object* objp, dock_function_info* infop)
+{
+	matrix local_moi, unorient, temp, world_moi;
+	// The MOI for a compound object is simply the sum of the MOI's of the parts, but
+	// they all have to be with respect to the same point (the center of mass, in this case).
+	// So for each part:
+
+	// We invert the inverse MOI to get an MOI in the local frame
+	vm_inverse_matrix(&local_moi, &objp->phys_info.I_body_inv);
+
+	// We calculate the inverse of the orientation matrix (which is also the transpose)
+	vm_copy_transpose(&unorient, &objp->orient);
+
+	// We calculate the world space MOI using (world MOI) = O^-1 * (local MOI) * O
+	// where O is the orientation matrix (which translates between local space and world space).
+	// Note that this is reverse of the usual convention because FS stores orientation matrices
+	// transposed relative to the linear algebra convention.
+	vm_matrix_x_matrix(&temp, &objp->orient, &local_moi);
+	vm_matrix_x_matrix(&world_moi, &temp, &unorient);
```

Comment:
Do you mean that the *formula* is the reverse of the usual convention, or the *code* is the reverse of the formula?  It looks like the latter, but it would be good to clarify in the comments.